lib/utils/documentdb_model_versioned.ts

```diff
@@ -94,9 +99,9 @@ export abstract class DocumentDbModelVersioned<
   public async update(
     objectId: string,
     partitionKey: string,
-    f: (current: T) => T
+    f: (current: T) => T | TR
```

Comment:
why `TR`? `TR` is `T` + storage specific attributes, the update functions shouldn't have any knowledge about them

Reply:
because TR has an 'id'. when we update the document it's a retrieved-document (or you have to get rid of the id before updating it, which is unpractical)

Reply:
in fact that's what happens, the ID get's stripped from the retrieved document by `toBaseType`

Reply:
ok